grpc/grpcserver.go

```diff
@@ -0,0 +1,60 @@
+/*
+ Copyright 2019 Vimeo Inc.
+ Adapted from https://github.com/charithe/gcgrpcpool/blob/master/gcgrpcpool.go
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package grpcpeerconn
+
+import (
+	"context"
+	"fmt"
+
+	gc "github.com/vimeo/galaxycache"
+	pb "github.com/vimeo/galaxycache/galaxycachepb"
+
+	"google.golang.org/grpc"
+)
+
+// GRPCGalaxyCacheServer implements the GalaxyCacheServer
+// interface generated by the GalaxyCache pb service
+type GRPCGalaxyCacheServer struct {
+	universe *gc.Universe
+}
+
+// RegisterGRPCServer registers the given grpc.Server with
+// a Universe for GetFromPeer calls over RPC
+func RegisterGRPCServer(universe *gc.Universe, grpcServer *grpc.Server) {
+	pb.RegisterGalaxyCacheServer(grpcServer, &GRPCGalaxyCacheServer{universe: universe})
+}
+
+// GetFromPeer implements the generated GalaxyCacheServer
+// interface, making an internal Get() after receiving a
+// remote call from a peer
+func (gp *GRPCGalaxyCacheServer) GetFromPeer(ctx context.Context, req *pb.GetRequest) (*pb.GetResponse, error) {
+	galaxy := gp.universe.GetGalaxy(req.Galaxy)
+	if galaxy == nil {
+		return nil, fmt.Errorf("Unable to find galaxy [%s]", req.Galaxy)
```

Comment:
Let's set an appropriate status code here: [`codes.InvalidArgument`](https://godoc.org/google.golang.org/grpc/codes#InvalidArgument)
```suggestion
		return nil, status.Errorf(codes.InvalidArgument, "Unable to find galaxy [%s]", req.Galaxy)
```

(see below for more detail about `status`)